### PR TITLE
 add README.zh-CN.md

### PR DESCRIPTION
**Pull request by hqc19907228**
translate README.md to README.zh-CN.md

### CODE REVIEW

File: README.zh-CN.md [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+}
+```
+
+更多示例可以在 [go-storage-example](https://github.com/beyondstorage/go-storage-example) 找到。
+
+## 特点
+
+### 支持多种原生服务
+
+目前已经有 **14** 个稳定的服务通过了所有的 [集成测试](https://github.com/beyondstorage/go-integration-test)。
```
Comment (abyss-w) [inline README.zh-CN.md:86]: I've added two more services, so maybe we should add them in.

File: README.zh-CN.md [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+- [gcs](https://github.com/beyondstorage/go-service-gcs/): [Google Cloud Storage](https://cloud.google.com/storage/)
+- [kodo](https://github.com/beyondstorage/go-service-kodo/): [qiniu kodo](https://www.qiniu.com/products/kodo)
+- [oss](https://github.com/beyondstorage/go-service-oss/): [Aliyun Object Storage](https://www.aliyun.com/product/oss)
+- [qingstor](https://github.com/beyondstorage/go-service-qingstor/): [QingStor Object Storage](https://www.qingcloud.com/products/qingstor/)
+- [s3](https://github.com/beyondstorage/go-service-s3/): [Amazon S3](https://aws.amazon.com/s3/)
+- [ftp](https://github.com/beyondstorage/go-service-ftp/): FTP
+- [gdrive](https://github.com/beyondstorage/go-service-gdrive): [Google Drive](https://www.google.com/drive/)
+- [ipfs](https://github.com/beyondstorage/go-service-ipfs): [InterPlanetary File System](https://ipfs.io)
+- [memory](https://github.com/beyondstorage/go-service-memory): data that only in memory
+- [minio](https://github.com/beyondstorage/go-service-minio): [MinIO](https://min.io)
```
Comment (abyss-w) [inline README.zh-CN.md:101]: My last PR sorted these services alphabetically. Maybe you can refer to it.

File: README.zh-CN.md [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+
+另有 **3** 个公测版本的服务已实现了所需功能，但还没有通过 [集成测试](https://github.com/beyondstorage/go-integration-test)。
+
+- [uss](https://github.com/beyondstorage/go-service-uss/): [UPYUN Storage Service](https://www.upyun.com/products/file-storage)
+- [hdfs](https://github.com/beyondstorage/go-service-hdfs): [Hadoop Distributed File System](https://hadoop.apache.org/docs/r1.2.1/hdfs_design.html#Introduction)
+- [tar](https://github.com/beyondstorage/go-service-tar): tar files
```
Comment (abyss-w) [inline README.zh-CN.md:107]: ditto

File: README.zh-CN.md [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+o, err := ps.CreatePage("path")
+// Write 1024 bytes from io.Reader at offset 2048
+n, err := ps.WritePage(o, r, 1024, 2048)
+```
+
+### 全面广泛的元数据
+
+全局对象元数据
+
+- `id`: unique key in service
```
Comment (abyss-w) [inline README.zh-CN.md:230]: Did you forget to translate here.

File: README.zh-CN.md [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+- `mode`: 对象的模式可以由以下几种进行组合：`read`, `dir`, `part` and [more](https://github.com/beyondstorage/go-storage/blob/master/types/object.go#L11) 
+- `etag`: [rfc2616](https://tools.ietf.org/html/rfc2616#section-14.19) 中定义的实体标签
+- `content-length`: 对象的内容大小
+- `content-md5`: [rfc2616](https://tools.ietf.org/html/rfc2616#section-14.15) 中定义的 Md5 简介
+- `content-type`: [rfc2616](https://tools.ietf.org/html/rfc2616#section-14.17) 中定义的媒体类型
+- `last-modified`: 对象上次更新的时间
+
+系统对象元数据
+
+Service system object metadata like `storage-class` and so on.
```
Comment (abyss-w) [inline README.zh-CN.md:241]: ditto

File: README.zh-CN.md [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+一个**厂商中立**的存储库。
+
+## 愿景
+
+**只需一次编写，即可在任意存储服务中运行。**
+
+## 目标
+
+- 厂商中立
+- 随时投产
```
Comment (Xuanwo) [inline README.zh-CN.md:19]: 翻译成 `生产就绪` 如何？

Comment (hqc19907228) [reply README.zh-CN.md:19]: ok

File: README.zh-CN.md [hunk 7]
```diff
@@ -0,0 +1,311 @@
...
+    // 添加 gcs 支持
+    _ "github.com/beyondstorage/go-service-gcs/v2"
+    // 添加 azblob 支持
+    _ "github.com/beyondstorage/go-service-azblob/v2"
+    // 更多支持，可在 BeyondStorage 下获取
+    _ "github.com/beyondstorage/go-service-xxx" 
+)
+
+func main() {
+    // 连接字符串并初始化存储器
```
Comment (JinnyYi) [inline README.zh-CN.md:46]: How about "使用`connection string`初始化`storager`" or "使用连接字符串初始化存储器"？
I'm not sure if some of the unique terms need to be translated as well？

File: README.zh-CN.md [hunk 8]
```diff
@@ -0,0 +1,311 @@
...
+    // 从 hello.txt 中读取数据到 io.Writer
+    n, err := store.Read("hello.txt", w)
+
+    // 检查 hello.txt 是否存在并获取其元数据
+    o, err := store.Stat("hello.txt")
+
+    // 使用对象的函数获取元数据
+    length, ok := o.GetContentLength()
+    
+    // 列表将在路径下创建一个对象的迭代器
```
Comment (JinnyYi) [inline README.zh-CN.md:64]: How about "`List`将创建路径下对象的迭代器" or "列举将创建路径下对象的迭代器"？

File: README.zh-CN.md [hunk 9]
```diff
@@ -0,0 +1,311 @@
...
+bs := store.(Blocker)
+
+// 创建一个块对象
+o, err := bs.CreateBlock("path")
+// 将 io.reader 中的 1024 字节写入 id 为 ”id-abc“ 的块
+n, err := bs.WriteBlock(o, r, 1024, "id-abc")
+// 通过块 id 组合区块
+err := bs.CombineBlock(o, []string{"id-abc"})
+```
+- 页面：允许进行随机写入
```
Comment (JinnyYi) [inline README.zh-CN.md:216]: How about "`页`"， or keep `Pager` without translation?

File: README.zh-CN.md [hunk 10]
```diff
@@ -0,0 +1,311 @@
...
+n, err := ps.WritePage(o, r, 1024, 2048)
+```
+
+### 综合元数据
+
+全局对象元数据
+
+- `id`: 服务中的唯一键
+- `name`: 服务工作目录的相对路径
+- `mode`: 对象的模式可以由以下几种进行组合：`read`, `dir`, `part` and [more](https://github.com/beyondstorage/go-storage/blob/master/types/object.go#L11)
```
Comment (JinnyYi) [inline README.zh-CN.md:232]: Not fully translated.

File: README.zh-CN.md [hunk 11]
```diff
@@ -0,0 +1,311 @@
...
+        return o.contentMd5, true
+    }
+    
+    return "", false
+}
+```
+
+### 服务器端加密
+
+服务器端加密支持在 system pair 和 system metadata 中使用, 并且我们可以通过 [Default Pairs](https://beyondstorage.io/docs/go-storage/pairs/index#default-pairs) 来简化工作。
```
Comment (JinnyYi) [inline README.zh-CN.md:283]: How about "支持通过 system pair 和 system metadata 指定在服务端对数据进行加密编码"？

File: README.zh-CN.md [hunk 12]
```diff
@@ -0,0 +1,311 @@
...
+            // 需要, 必须为 AES256
+            s3.WithServerSideEncryptionCustomerAlgorithm(s3.ServerSideEncryptionAes256),
+            // 你的 AES-256 密钥, 需要是一个 32 字节的二进制值
+            s3.WithServerSideEncryptionCustomerKey(key),
+        },
+        // 现在你必须提供客户密钥才能读取加密数据
+        Read: []types.Pair{
+            // 需要, 必须为 AES256
+            s3.WithServerSideEncryptionCustomerAlgorithm(s3.ServerSideEncryptionAes256),
+            // Required, your AES-256 key, a 32-byte binary value
```
Comment (JinnyYi) [inline README.zh-CN.md:299]: Need to be translated.

Comment (hqc19907228) [reply README.zh-CN.md:299]: Required, your AES-256 key, a 32-byte binary value
这个没明白是什么意思。。

**Comment by Xuanwo** [comment]
ping @abyss-w to take another look.

File: README.zh-CN.md [hunk 13]
```diff
@@ -0,0 +1,313 @@
...
+- [uss](https://github.com/beyondstorage/go-service-uss/): [UPYUN Storage Service](https://www.upyun.com/products/file-storage)
+
+最后还有 **4** 个处于内测阶段的服务仍在开发中。
+
+- [onedrive](https://github.com/beyondstorage/go-service-onedrive): [Microsoft OneDrive](https://www.microsoft.com/en-ww/microsoft-365/onedrive/online-cloud-storage)
+- [storj](https://github.com/beyondstorage/go-service-storj): [StorJ](https://www.storj.io/)
+- [webdav](https://github.com/beyondstorage/go-service-webdav): [WebDAV](http://www.webdav.org/)
+- [zip](https://github.com/beyondstorage/go-service-zip): zip files
+
+更多关于服务的想法可以在 [Service Integration Tracking](https://github.com/beyondstorage/go-storage/issues/536) 找到。
```
Comment (abyss-w) [inline README.zh-CN.md:118]: Does `Service Integration Tracking` need to be translated?

Comment (Xuanwo) [reply README.zh-CN.md:118]: It's OK to keep it.

**Review by abyss-w — APPROVED** [review]
LGTM

**Pull request merged**